compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -101,7 +101,24 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
 
           // (fun)[args]
           case TypeApply(fun, args) =>
-            cpy.TypeApply(tree)(transform(fun), args)
+            val tfun = transform(fun)
+            tfun match
+              case InstrumentCoverage.InstrumentedBlock(invokeCall, expr) =>
```

Comment:
Using an extractor after the fact does not seem consistent with how you handled `Ident`s and `Select`s using `canInstrumentParameterless`. Is there a good reason you can't do something similar here?

Reply:
Doing so would introduce more complexity, I think, because I would have to handle:
- `TypeApply(Apply(...`
- `TypeApply(Select(...`
- TypeApply(Ident(...`

with all the conditions and details of `Apply`, `Select` and `Ident`. Those include the lifting of `Apply` arguments, which calls `instrumentLifted`, which would require a refactor in order to move the `invokeCall` outside of the `TypeApply`.

Reply:
Hum, OK. Let's take it as is. But I feel that there are opportunities for simplifications starting to pop up in this phase.